src/main/java/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package bookstore.service.impl;
+
+import bookstore.dto.BookDto;
+import bookstore.dto.BookSearchParametersDto;
+import bookstore.dto.CreateBookRequestDto;
+import bookstore.entity.Book;
+import bookstore.exception.EntityNotFoundException;
+import bookstore.mapper.BookMapper;
+import bookstore.repository.book.BookRepository;
+import bookstore.repository.book.BookSpecificationBuilder;
+import bookstore.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private final BookSpecificationBuilder specificationBuilder;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toBook(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll()
+                .stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getById(Long id) {
+        return bookRepository.findById(id)
+                .map(bookMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException(
+                                "Can't find book with id " + id));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public List<BookDto> searchBookByParams(BookSearchParametersDto parameters) {
+        Specification<Book> bookSpecification = specificationBuilder.build(parameters);
+        return bookRepository.findAll(bookSpecification)
+                .stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toBook(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```

Comment:
bad approach

you should fetch the desired book by id, then update it, and save afterwards

you can leverage mapstruct to perform update:
`void updateBook(CreateBookRequestDto  dto, @MappingTarget Book book);`

the final version will be like:
```
Book book = repo.findById(id).orElseThrow(...);
mapper.updateBook(dto, book);
return mapper.toDto(repo.save(book));
```